{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c77f8da1-2837-4bf7-b3f4-dfcaa7ab42f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-08T23:28:20.070699Z",
     "iopub.status.busy": "2023-11-08T23:28:20.070554Z",
     "iopub.status.idle": "2023-11-08T23:28:20.158878Z",
     "shell.execute_reply": "2023-11-08T23:28:20.158185Z"
    }
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import numpy as np\n",
    "import random\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3fe25376-b856-4296-ba02-5e3771ff9451",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-11-08T23:28:20.160911Z",
     "iopub.status.busy": "2023-11-08T23:28:20.160751Z",
     "iopub.status.idle": "2023-11-09T01:23:28.683549Z",
     "shell.execute_reply": "2023-11-09T01:23:28.682867Z"
    },
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:28:20.346743: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "/usr/lib/python3.11/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.2 when it was built against 1.14.1, this may cause problems\n",
      "  _warn((\"h5py is running against HDF5 {0} when it was built against {1}, \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35935\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:29:43.887836: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:43.899406: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:43.899644: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:43.903163: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:43.903355: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:43.903524: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:43.973479: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:43.973685: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:43.973851: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:43.973994: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 18:29:43.974346: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:29:44.900280: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:44.900521: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:44.900694: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:44.900906: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:44.901083: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:29:44.901221: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 18:29:44.901252: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n",
      "2023-11-08 18:29:44.926496: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:375] MLIR V1 optimization pass is not enabled\n",
      "2023-11-08 18:29:45.147075: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_20/lstm_cell_20/bias/Assign' id:3345 op device:{requested: '', assigned: ''} def:{{{node lstm_20/lstm_cell_20/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_20/lstm_cell_20/bias, lstm_20/lstm_cell_20/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:29:45.282775: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_1' id:3634 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_1}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 0 0 0...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:29:45.317018: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_2' id:3635 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_2}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 1 1 1...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "(32325, 95)\n",
      "Train on 32325 samples, validate on 3610 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:29:49.757408: W tensorflow/c/c_api.cc:304] Operation '{name:'training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v/Assign' id:17552 op device:{requested: '', assigned: ''} def:{{{node training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v, training/Adam/lstm_34/lstm_cell_34/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:29:53.654455: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n",
      "2023-11-08 18:29:55.897251: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "2023-11-08 18:29:55.910658: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8902\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32325/32325 [==============================] - ETA: 0s - loss: 2.7967"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2335: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates = self.state_updates\n",
      "2023-11-08 18:30:18.534589: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_1/mul' id:6477 op device:{requested: '', assigned: ''} def:{{{node loss_1/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_1/mul/x, loss_1/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.80412, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 30s 931us/sample - loss: 2.7967 - val_loss: 1.8041\n",
      "Epoch 2/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.7562\n",
      "Epoch 2: val_loss improved from 1.80412 to 1.53836, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 717us/sample - loss: 1.7562 - val_loss: 1.5384\n",
      "Epoch 3/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.5954\n",
      "Epoch 3: val_loss improved from 1.53836 to 1.46202, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 722us/sample - loss: 1.5954 - val_loss: 1.4620\n",
      "Epoch 4/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.5401\n",
      "Epoch 4: val_loss improved from 1.46202 to 1.43645, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 721us/sample - loss: 1.5401 - val_loss: 1.4365\n",
      "Epoch 5/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.5164\n",
      "Epoch 5: val_loss improved from 1.43645 to 1.42285, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 719us/sample - loss: 1.5164 - val_loss: 1.4229\n",
      "Epoch 6/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.5023\n",
      "Epoch 6: val_loss improved from 1.42285 to 1.41596, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 725us/sample - loss: 1.5023 - val_loss: 1.4160\n",
      "Epoch 7/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4903\n",
      "Epoch 7: val_loss improved from 1.41596 to 1.40097, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 717us/sample - loss: 1.4903 - val_loss: 1.4010\n",
      "Epoch 8/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4851\n",
      "Epoch 8: val_loss did not improve from 1.40097\n",
      "32325/32325 [==============================] - 23s 717us/sample - loss: 1.4851 - val_loss: 1.4041\n",
      "Epoch 9/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4845\n",
      "Epoch 9: val_loss improved from 1.40097 to 1.39264, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 717us/sample - loss: 1.4845 - val_loss: 1.3926\n",
      "Epoch 10/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4798\n",
      "Epoch 10: val_loss improved from 1.39264 to 1.38965, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 721us/sample - loss: 1.4798 - val_loss: 1.3896\n",
      "Epoch 11/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4810\n",
      "Epoch 11: val_loss did not improve from 1.38965\n",
      "32325/32325 [==============================] - 20s 624us/sample - loss: 1.4810 - val_loss: 1.3929\n",
      "Epoch 12/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4803\n",
      "Epoch 12: val_loss did not improve from 1.38965\n",
      "32325/32325 [==============================] - 23s 705us/sample - loss: 1.4803 - val_loss: 1.3912\n",
      "Epoch 13/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4733\n",
      "Epoch 13: val_loss improved from 1.38965 to 1.38397, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 20s 623us/sample - loss: 1.4733 - val_loss: 1.3840\n",
      "Epoch 14/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4727\n",
      "Epoch 14: val_loss did not improve from 1.38397\n",
      "32325/32325 [==============================] - 21s 649us/sample - loss: 1.4727 - val_loss: 1.3919\n",
      "Epoch 15/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4698\n",
      "Epoch 15: val_loss improved from 1.38397 to 1.38363, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 22s 690us/sample - loss: 1.4698 - val_loss: 1.3836\n",
      "Epoch 16/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4749\n",
      "Epoch 16: val_loss did not improve from 1.38363\n",
      "32325/32325 [==============================] - 20s 624us/sample - loss: 1.4749 - val_loss: 1.3865\n",
      "Epoch 17/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4910\n",
      "Epoch 17: val_loss did not improve from 1.38363\n",
      "32325/32325 [==============================] - 20s 626us/sample - loss: 1.4910 - val_loss: 1.4007\n",
      "Epoch 18/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4982\n",
      "Epoch 18: val_loss did not improve from 1.38363\n",
      "32325/32325 [==============================] - 20s 628us/sample - loss: 1.4982 - val_loss: 1.3889\n",
      "Epoch 19/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4881\n",
      "Epoch 19: val_loss did not improve from 1.38363\n",
      "32325/32325 [==============================] - 20s 623us/sample - loss: 1.4881 - val_loss: 1.3891\n",
      "Epoch 20/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4754\n",
      "Epoch 20: val_loss did not improve from 1.38363\n",
      "32325/32325 [==============================] - 20s 624us/sample - loss: 1.4754 - val_loss: 1.3888\n",
      "Epoch 21/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4757\n",
      "Epoch 21: val_loss did not improve from 1.38363\n",
      "32325/32325 [==============================] - 21s 643us/sample - loss: 1.4757 - val_loss: 1.3902\n",
      "Epoch 22/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4619\n",
      "Epoch 22: val_loss improved from 1.38363 to 1.38312, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 21s 657us/sample - loss: 1.4619 - val_loss: 1.3831\n",
      "Epoch 23/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4545\n",
      "Epoch 23: val_loss did not improve from 1.38312\n",
      "32325/32325 [==============================] - 23s 718us/sample - loss: 1.4545 - val_loss: 1.3835\n",
      "Epoch 24/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4535\n",
      "Epoch 24: val_loss did not improve from 1.38312\n",
      "32325/32325 [==============================] - 22s 683us/sample - loss: 1.4535 - val_loss: 1.3859\n",
      "Epoch 25/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4480\n",
      "Epoch 25: val_loss improved from 1.38312 to 1.37588, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 20s 630us/sample - loss: 1.4480 - val_loss: 1.3759\n",
      "Epoch 26/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4463\n",
      "Epoch 26: val_loss improved from 1.37588 to 1.37257, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 20s 633us/sample - loss: 1.4463 - val_loss: 1.3726\n",
      "Epoch 27/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4416\n",
      "Epoch 27: val_loss improved from 1.37257 to 1.37043, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 21s 639us/sample - loss: 1.4416 - val_loss: 1.3704\n",
      "Epoch 28/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4379\n",
      "Epoch 28: val_loss did not improve from 1.37043\n",
      "32325/32325 [==============================] - 20s 626us/sample - loss: 1.4379 - val_loss: 1.3755\n",
      "Epoch 29/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4347\n",
      "Epoch 29: val_loss did not improve from 1.37043\n",
      "32325/32325 [==============================] - 20s 625us/sample - loss: 1.4347 - val_loss: 1.3721\n",
      "Epoch 30/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4340\n",
      "Epoch 30: val_loss improved from 1.37043 to 1.36313, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 21s 646us/sample - loss: 1.4340 - val_loss: 1.3631\n",
      "Epoch 31/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4324\n",
      "Epoch 31: val_loss did not improve from 1.36313\n",
      "32325/32325 [==============================] - 20s 629us/sample - loss: 1.4324 - val_loss: 1.3678\n",
      "Epoch 32/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4300\n",
      "Epoch 32: val_loss did not improve from 1.36313\n",
      "32325/32325 [==============================] - 23s 709us/sample - loss: 1.4300 - val_loss: 1.3662\n",
      "Epoch 33/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4265\n",
      "Epoch 33: val_loss improved from 1.36313 to 1.35799, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 22s 674us/sample - loss: 1.4265 - val_loss: 1.3580\n",
      "Epoch 34/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4281\n",
      "Epoch 34: val_loss did not improve from 1.35799\n",
      "32325/32325 [==============================] - 22s 691us/sample - loss: 1.4281 - val_loss: 1.3665\n",
      "Epoch 35/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4236\n",
      "Epoch 35: val_loss did not improve from 1.35799\n",
      "32325/32325 [==============================] - 23s 719us/sample - loss: 1.4236 - val_loss: 1.3612\n",
      "Epoch 36/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4205\n",
      "Epoch 36: val_loss improved from 1.35799 to 1.35749, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 721us/sample - loss: 1.4205 - val_loss: 1.3575\n",
      "Epoch 37/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4208\n",
      "Epoch 37: val_loss improved from 1.35749 to 1.35179, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 723us/sample - loss: 1.4208 - val_loss: 1.3518\n",
      "Epoch 38/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4212\n",
      "Epoch 38: val_loss did not improve from 1.35179\n",
      "32325/32325 [==============================] - 23s 708us/sample - loss: 1.4212 - val_loss: 1.3588\n",
      "Epoch 39/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4177\n",
      "Epoch 39: val_loss improved from 1.35179 to 1.35143, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 721us/sample - loss: 1.4177 - val_loss: 1.3514\n",
      "Epoch 40/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4163\n",
      "Epoch 40: val_loss did not improve from 1.35143\n",
      "32325/32325 [==============================] - 23s 720us/sample - loss: 1.4163 - val_loss: 1.3629\n",
      "Epoch 41/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4114\n",
      "Epoch 41: val_loss did not improve from 1.35143\n",
      "32325/32325 [==============================] - 23s 718us/sample - loss: 1.4114 - val_loss: 1.3542\n",
      "Epoch 42/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4124\n",
      "Epoch 42: val_loss did not improve from 1.35143\n",
      "32325/32325 [==============================] - 23s 716us/sample - loss: 1.4124 - val_loss: 1.3567\n",
      "Epoch 43/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4101\n",
      "Epoch 43: val_loss did not improve from 1.35143\n",
      "32325/32325 [==============================] - 23s 712us/sample - loss: 1.4101 - val_loss: 1.3628\n",
      "Epoch 44/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4096\n",
      "Epoch 44: val_loss did not improve from 1.35143\n",
      "32325/32325 [==============================] - 23s 713us/sample - loss: 1.4096 - val_loss: 1.3517\n",
      "Epoch 45/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4068\n",
      "Epoch 45: val_loss did not improve from 1.35143\n",
      "32325/32325 [==============================] - 23s 713us/sample - loss: 1.4068 - val_loss: 1.3706\n",
      "Epoch 46/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4057\n",
      "Epoch 46: val_loss did not improve from 1.35143\n",
      "32325/32325 [==============================] - 23s 714us/sample - loss: 1.4057 - val_loss: 1.3618\n",
      "Epoch 47/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4073\n",
      "Epoch 47: val_loss improved from 1.35143 to 1.35074, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 720us/sample - loss: 1.4073 - val_loss: 1.3507\n",
      "Epoch 48/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4052\n",
      "Epoch 48: val_loss improved from 1.35074 to 1.34784, saving model to ./checkpoints/unknown_person_baseline_p27_20.h5\n",
      "32325/32325 [==============================] - 22s 689us/sample - loss: 1.4052 - val_loss: 1.3478\n",
      "Epoch 49/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4028\n",
      "Epoch 49: val_loss did not improve from 1.34784\n",
      "32325/32325 [==============================] - 21s 642us/sample - loss: 1.4028 - val_loss: 1.3596\n",
      "Epoch 50/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4008\n",
      "Epoch 50: val_loss did not improve from 1.34784\n",
      "32325/32325 [==============================] - 21s 642us/sample - loss: 1.4008 - val_loss: 1.3558\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:48:24.200678: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_16_1/lstm_cell_53/kernel/Assign' id:21200 op device:{requested: '', assigned: ''} def:{{{node lstm_16_1/lstm_cell_53/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_16_1/lstm_cell_53/kernel, lstm_16_1/lstm_cell_53/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:48:25.924208: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_37/lstm_cell_37/bias/v/Assign' id:25905 op device:{requested: '', assigned: ''} def:{{{node lstm_37/lstm_cell_37/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_37/lstm_cell_37/bias/v, lstm_37/lstm_cell_37/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2359: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates=self.state_updates,\n",
      "2023-11-08 18:48:27.298526: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_1/cond/Merge' id:24591 op device:{requested: '', assigned: ''} def:{{{node dropout_4_1/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_1/cond/Identity, dropout_4_1/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1066)\n",
      "(1514, 1066)\n",
      "(1644, 1066)\n",
      "(1764, 1066)\n",
      "(1836, 1066)\n",
      "(1699, 1066)\n",
      "(1369, 1066)\n",
      "(1766, 1066)\n",
      "(1631, 1066)\n",
      "(1692, 1066)\n",
      "(1550, 1066)\n",
      "(1920, 1066)\n",
      "(1739, 1066)\n",
      "(1788, 1066)\n",
      "(1812, 1066)\n",
      "(1788, 1066)\n",
      "(1788, 1066)\n",
      "(1668, 1066)\n",
      "(1872, 1066)\n",
      "{1: 6.034565399156354, 2: 6.457919217457563, 4: 10.0, 5: 5.188280737211687, 6: 5.354961400051734, 8: 8.753628496246451, 9: 9.142561277259889, 10: 7.1913218856805665, 11: 7.9880696430863445, 12: 8.667195057319244, 13: 6.491059534666087, 17: 9.310379421749268, 19: 8.392628894224599, 21: 8.01390817068516, 22: 1.0, 25: 7.606949254556316, 26: 5.999092823277199, 28: 5.433351579849667, 29: 2.3630493158679813}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2793144/859520161.py:300: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)\n",
      "  w_train[i] = weights_dict[int(p_train[i])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 32325 samples, validate on 3610 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:52:29.942029: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_5/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32325/32325 [==============================] - ETA: 0s - loss: 9.9288\n",
      "Epoch 1: val_loss improved from inf to 1.37793, saving model to ./checkpoints/unknown_person_p27_20.h5\n",
      "32325/32325 [==============================] - 28s 853us/sample - loss: 9.9288 - val_loss: 1.3779\n",
      "Epoch 2/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.8571\n",
      "Epoch 2: val_loss improved from 1.37793 to 1.36511, saving model to ./checkpoints/unknown_person_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 719us/sample - loss: 9.8571 - val_loss: 1.3651\n",
      "Epoch 3/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.7942\n",
      "Epoch 3: val_loss improved from 1.36511 to 1.36269, saving model to ./checkpoints/unknown_person_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 707us/sample - loss: 9.7942 - val_loss: 1.3627\n",
      "Epoch 4/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.7850\n",
      "Epoch 4: val_loss did not improve from 1.36269\n",
      "32325/32325 [==============================] - 21s 644us/sample - loss: 9.7850 - val_loss: 1.3652\n",
      "Epoch 5/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.7274\n",
      "Epoch 5: val_loss improved from 1.36269 to 1.35808, saving model to ./checkpoints/unknown_person_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 723us/sample - loss: 9.7274 - val_loss: 1.3581\n",
      "Epoch 6/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.7005\n",
      "Epoch 6: val_loss improved from 1.35808 to 1.35462, saving model to ./checkpoints/unknown_person_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 710us/sample - loss: 9.7005 - val_loss: 1.3546\n",
      "Epoch 7/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.6989\n",
      "Epoch 7: val_loss did not improve from 1.35462\n",
      "32325/32325 [==============================] - 23s 723us/sample - loss: 9.6989 - val_loss: 1.3665\n",
      "Epoch 8/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.6694\n",
      "Epoch 8: val_loss improved from 1.35462 to 1.34282, saving model to ./checkpoints/unknown_person_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 721us/sample - loss: 9.6694 - val_loss: 1.3428\n",
      "Epoch 9/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.6644\n",
      "Epoch 9: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 23s 718us/sample - loss: 9.6644 - val_loss: 1.3638\n",
      "Epoch 10/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.6397\n",
      "Epoch 10: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 23s 699us/sample - loss: 9.6397 - val_loss: 1.3556\n",
      "Epoch 11/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.6403\n",
      "Epoch 11: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 21s 646us/sample - loss: 9.6403 - val_loss: 1.3551\n",
      "Epoch 12/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.6122\n",
      "Epoch 12: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 20s 627us/sample - loss: 9.6122 - val_loss: 1.3479\n",
      "Epoch 13/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.6082\n",
      "Epoch 13: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 22s 687us/sample - loss: 9.6082 - val_loss: 1.3526\n",
      "Epoch 14/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5914\n",
      "Epoch 14: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 21s 658us/sample - loss: 9.5914 - val_loss: 1.3717\n",
      "Epoch 15/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5959\n",
      "Epoch 15: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 23s 717us/sample - loss: 9.5959 - val_loss: 1.3553\n",
      "Epoch 16/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5849\n",
      "Epoch 16: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 23s 717us/sample - loss: 9.5849 - val_loss: 1.3503\n",
      "Epoch 17/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5628\n",
      "Epoch 17: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 23s 700us/sample - loss: 9.5628 - val_loss: 1.3555\n",
      "Epoch 18/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5525\n",
      "Epoch 18: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 21s 658us/sample - loss: 9.5525 - val_loss: 1.3540\n",
      "Epoch 19/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5351\n",
      "Epoch 19: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 21s 660us/sample - loss: 9.5351 - val_loss: 1.3546\n",
      "Epoch 20/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5260\n",
      "Epoch 20: val_loss did not improve from 1.34282\n",
      "32325/32325 [==============================] - 20s 617us/sample - loss: 9.5260 - val_loss: 1.3514\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:00:01.142824: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_8_2/lstm_cell_82/bias/Assign' id:39347 op device:{requested: '', assigned: ''} def:{{{node lstm_8_2/lstm_cell_82/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_8_2/lstm_cell_82/bias, lstm_8_2/lstm_cell_82/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 19:00:03.500858: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_35_2/lstm_cell_109/bias/m/Assign' id:45185 op device:{requested: '', assigned: ''} def:{{{node lstm_35_2/lstm_cell_109/bias/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_35_2/lstm_cell_109/bias/m, lstm_35_2/lstm_cell_109/bias/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 32325 samples, validate on 3610 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:00:08.536806: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:00:20.798524: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_6/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4028"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:00:41.929026: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.34939, saving model to ./checkpoints/unknown_person_baseline_2_p27_20.h5\n",
      "32325/32325 [==============================] - 27s 844us/sample - loss: 1.4028 - val_loss: 1.3494\n",
      "Epoch 2/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4012\n",
      "Epoch 2: val_loss did not improve from 1.34939\n",
      "32325/32325 [==============================] - 23s 712us/sample - loss: 1.4012 - val_loss: 1.3497\n",
      "Epoch 3/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4006\n",
      "Epoch 3: val_loss did not improve from 1.34939\n",
      "32325/32325 [==============================] - 21s 645us/sample - loss: 1.4006 - val_loss: 1.3502\n",
      "Epoch 4/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3996\n",
      "Epoch 4: val_loss did not improve from 1.34939\n",
      "32325/32325 [==============================] - 20s 624us/sample - loss: 1.3996 - val_loss: 1.3499\n",
      "Epoch 5/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4007\n",
      "Epoch 5: val_loss improved from 1.34939 to 1.34231, saving model to ./checkpoints/unknown_person_baseline_2_p27_20.h5\n",
      "32325/32325 [==============================] - 20s 633us/sample - loss: 1.4007 - val_loss: 1.3423\n",
      "Epoch 6/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3965\n",
      "Epoch 6: val_loss did not improve from 1.34231\n",
      "32325/32325 [==============================] - 22s 672us/sample - loss: 1.3965 - val_loss: 1.3471\n",
      "Epoch 7/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3964\n",
      "Epoch 7: val_loss did not improve from 1.34231\n",
      "32325/32325 [==============================] - 23s 715us/sample - loss: 1.3964 - val_loss: 1.3475\n",
      "Epoch 8/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3935\n",
      "Epoch 8: val_loss did not improve from 1.34231\n",
      "32325/32325 [==============================] - 21s 653us/sample - loss: 1.3935 - val_loss: 1.3450\n",
      "Epoch 9/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3922\n",
      "Epoch 9: val_loss improved from 1.34231 to 1.34207, saving model to ./checkpoints/unknown_person_baseline_2_p27_20.h5\n",
      "32325/32325 [==============================] - 20s 631us/sample - loss: 1.3922 - val_loss: 1.3421\n",
      "Epoch 10/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3898\n",
      "Epoch 10: val_loss did not improve from 1.34207\n",
      "32325/32325 [==============================] - 20s 627us/sample - loss: 1.3898 - val_loss: 1.3494\n",
      "Epoch 11/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3893\n",
      "Epoch 11: val_loss did not improve from 1.34207\n",
      "32325/32325 [==============================] - 22s 680us/sample - loss: 1.3893 - val_loss: 1.3514\n",
      "Epoch 12/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3878\n",
      "Epoch 12: val_loss improved from 1.34207 to 1.34191, saving model to ./checkpoints/unknown_person_baseline_2_p27_20.h5\n",
      "32325/32325 [==============================] - 23s 707us/sample - loss: 1.3878 - val_loss: 1.3419\n",
      "Epoch 13/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3883\n",
      "Epoch 13: val_loss did not improve from 1.34191\n",
      "32325/32325 [==============================] - 20s 632us/sample - loss: 1.3883 - val_loss: 1.3437\n",
      "Epoch 14/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3865\n",
      "Epoch 14: val_loss improved from 1.34191 to 1.34161, saving model to ./checkpoints/unknown_person_baseline_2_p27_20.h5\n",
      "32325/32325 [==============================] - 22s 681us/sample - loss: 1.3865 - val_loss: 1.3416\n",
      "Epoch 15/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3841\n",
      "Epoch 15: val_loss did not improve from 1.34161\n",
      "32325/32325 [==============================] - 23s 713us/sample - loss: 1.3841 - val_loss: 1.3449\n",
      "Epoch 16/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3862\n",
      "Epoch 16: val_loss did not improve from 1.34161\n",
      "32325/32325 [==============================] - 21s 641us/sample - loss: 1.3862 - val_loss: 1.3428\n",
      "Epoch 17/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3812\n",
      "Epoch 17: val_loss did not improve from 1.34161\n",
      "32325/32325 [==============================] - 20s 628us/sample - loss: 1.3812 - val_loss: 1.3418\n",
      "Epoch 18/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3798\n",
      "Epoch 18: val_loss did not improve from 1.34161\n",
      "32325/32325 [==============================] - 20s 621us/sample - loss: 1.3798 - val_loss: 1.3483\n",
      "Epoch 19/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3810\n",
      "Epoch 19: val_loss did not improve from 1.34161\n",
      "32325/32325 [==============================] - 20s 623us/sample - loss: 1.3810 - val_loss: 1.3538\n",
      "Epoch 20/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3804\n",
      "Epoch 20: val_loss improved from 1.34161 to 1.33870, saving model to ./checkpoints/unknown_person_baseline_2_p27_20.h5\n",
      "32325/32325 [==============================] - 22s 691us/sample - loss: 1.3804 - val_loss: 1.3387\n",
      "Train on 32325 samples, validate on 3610 samples\n",
      "Epoch 1/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3775\n",
      "Epoch 1: val_loss improved from inf to 1.33964, saving model to ./checkpoints/unknown_person_baseline_p27_21.h5\n",
      "32325/32325 [==============================] - 23s 715us/sample - loss: 1.3775 - val_loss: 1.3396\n",
      "Epoch 2/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3764\n",
      "Epoch 2: val_loss did not improve from 1.33964\n",
      "32325/32325 [==============================] - 21s 643us/sample - loss: 1.3764 - val_loss: 1.3444\n",
      "Epoch 3/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3790\n",
      "Epoch 3: val_loss did not improve from 1.33964\n",
      "32325/32325 [==============================] - 20s 623us/sample - loss: 1.3790 - val_loss: 1.3476\n",
      "Epoch 4/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3767\n",
      "Epoch 4: val_loss improved from 1.33964 to 1.33572, saving model to ./checkpoints/unknown_person_baseline_p27_21.h5\n",
      "32325/32325 [==============================] - 23s 712us/sample - loss: 1.3767 - val_loss: 1.3357\n",
      "Epoch 5/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3765\n",
      "Epoch 5: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 22s 674us/sample - loss: 1.3765 - val_loss: 1.3408\n",
      "Epoch 6/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3730\n",
      "Epoch 6: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 23s 698us/sample - loss: 1.3730 - val_loss: 1.3434\n",
      "Epoch 7/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3727\n",
      "Epoch 7: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 21s 643us/sample - loss: 1.3727 - val_loss: 1.3577\n",
      "Epoch 8/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3744\n",
      "Epoch 8: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 21s 648us/sample - loss: 1.3744 - val_loss: 1.3392\n",
      "Epoch 9/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3824\n",
      "Epoch 9: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 23s 712us/sample - loss: 1.3824 - val_loss: 1.3364\n",
      "Epoch 10/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3847\n",
      "Epoch 10: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 23s 718us/sample - loss: 1.3847 - val_loss: 1.3369\n",
      "Epoch 11/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3842\n",
      "Epoch 11: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 23s 713us/sample - loss: 1.3842 - val_loss: 1.3374\n",
      "Epoch 12/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3973\n",
      "Epoch 12: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 22s 667us/sample - loss: 1.3973 - val_loss: 1.3474\n",
      "Epoch 13/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3990\n",
      "Epoch 13: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 628us/sample - loss: 1.3990 - val_loss: 1.3506\n",
      "Epoch 14/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4497\n",
      "Epoch 14: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 21s 658us/sample - loss: 1.4497 - val_loss: 1.3494\n",
      "Epoch 15/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4089\n",
      "Epoch 15: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 625us/sample - loss: 1.4089 - val_loss: 1.3553\n",
      "Epoch 16/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4009\n",
      "Epoch 16: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 628us/sample - loss: 1.4009 - val_loss: 1.3422\n",
      "Epoch 17/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4077\n",
      "Epoch 17: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 628us/sample - loss: 1.4077 - val_loss: 1.3550\n",
      "Epoch 18/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4231\n",
      "Epoch 18: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 622us/sample - loss: 1.4231 - val_loss: 1.3526\n",
      "Epoch 19/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4237\n",
      "Epoch 19: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 622us/sample - loss: 1.4237 - val_loss: 1.3633\n",
      "Epoch 20/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4222\n",
      "Epoch 20: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 627us/sample - loss: 1.4222 - val_loss: 1.3684\n",
      "Epoch 21/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4174\n",
      "Epoch 21: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 623us/sample - loss: 1.4174 - val_loss: 1.3644\n",
      "Epoch 22/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4265\n",
      "Epoch 22: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 626us/sample - loss: 1.4265 - val_loss: 1.3647\n",
      "Epoch 23/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4185\n",
      "Epoch 23: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 22s 686us/sample - loss: 1.4185 - val_loss: 1.3629\n",
      "Epoch 24/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4103\n",
      "Epoch 24: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 23s 715us/sample - loss: 1.4103 - val_loss: 1.3602\n",
      "Epoch 25/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4081\n",
      "Epoch 25: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 23s 703us/sample - loss: 1.4081 - val_loss: 1.3652\n",
      "Epoch 26/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4040\n",
      "Epoch 26: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 23s 710us/sample - loss: 1.4040 - val_loss: 1.3489\n",
      "Epoch 27/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4010\n",
      "Epoch 27: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 23s 711us/sample - loss: 1.4010 - val_loss: 1.3545\n",
      "Epoch 28/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4013\n",
      "Epoch 28: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 22s 681us/sample - loss: 1.4013 - val_loss: 1.3531\n",
      "Epoch 29/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3957\n",
      "Epoch 29: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 627us/sample - loss: 1.3957 - val_loss: 1.3524\n",
      "Epoch 30/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3990\n",
      "Epoch 30: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 623us/sample - loss: 1.3990 - val_loss: 1.3551\n",
      "Epoch 31/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3930\n",
      "Epoch 31: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 21s 637us/sample - loss: 1.3930 - val_loss: 1.3503\n",
      "Epoch 32/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3910\n",
      "Epoch 32: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 22s 686us/sample - loss: 1.3910 - val_loss: 1.3551\n",
      "Epoch 33/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3921\n",
      "Epoch 33: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 625us/sample - loss: 1.3921 - val_loss: 1.3460\n",
      "Epoch 34/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3905\n",
      "Epoch 34: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 625us/sample - loss: 1.3905 - val_loss: 1.3493\n",
      "Epoch 35/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3883\n",
      "Epoch 35: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 628us/sample - loss: 1.3883 - val_loss: 1.3478\n",
      "Epoch 36/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3903\n",
      "Epoch 36: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 22s 668us/sample - loss: 1.3903 - val_loss: 1.3499\n",
      "Epoch 37/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3872\n",
      "Epoch 37: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 21s 660us/sample - loss: 1.3872 - val_loss: 1.3461\n",
      "Epoch 38/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3857\n",
      "Epoch 38: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 616us/sample - loss: 1.3857 - val_loss: 1.3464\n",
      "Epoch 39/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3819\n",
      "Epoch 39: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 22s 668us/sample - loss: 1.3819 - val_loss: 1.3532\n",
      "Epoch 40/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3838\n",
      "Epoch 40: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 23s 698us/sample - loss: 1.3838 - val_loss: 1.3448\n",
      "Epoch 41/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3824\n",
      "Epoch 41: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 23s 701us/sample - loss: 1.3824 - val_loss: 1.3555\n",
      "Epoch 42/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3831\n",
      "Epoch 42: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 22s 673us/sample - loss: 1.3831 - val_loss: 1.3469\n",
      "Epoch 43/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3776\n",
      "Epoch 43: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 21s 657us/sample - loss: 1.3776 - val_loss: 1.3503\n",
      "Epoch 44/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3813\n",
      "Epoch 44: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 23s 703us/sample - loss: 1.3813 - val_loss: 1.3467\n",
      "Epoch 45/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3782\n",
      "Epoch 45: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 22s 672us/sample - loss: 1.3782 - val_loss: 1.3646\n",
      "Epoch 46/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3769\n",
      "Epoch 46: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 615us/sample - loss: 1.3769 - val_loss: 1.3457\n",
      "Epoch 47/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3765\n",
      "Epoch 47: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 618us/sample - loss: 1.3765 - val_loss: 1.3378\n",
      "Epoch 48/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3767\n",
      "Epoch 48: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 615us/sample - loss: 1.3767 - val_loss: 1.3438\n",
      "Epoch 49/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3754\n",
      "Epoch 49: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 617us/sample - loss: 1.3754 - val_loss: 1.3448\n",
      "Epoch 50/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3745\n",
      "Epoch 50: val_loss did not improve from 1.33572\n",
      "32325/32325 [==============================] - 20s 621us/sample - loss: 1.3745 - val_loss: 1.3438\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:25:18.963781: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_13_3/lstm_cell_124/kernel/Assign' id:59478 op device:{requested: '', assigned: ''} def:{{{node lstm_13_3/lstm_cell_124/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_13_3/lstm_cell_124/kernel, lstm_13_3/lstm_cell_124/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 19:25:22.197709: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_27_3/lstm_cell_138/recurrent_kernel/v/Assign' id:65063 op device:{requested: '', assigned: ''} def:{{{node lstm_27_3/lstm_cell_138/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_27_3/lstm_cell_138/recurrent_kernel/v, lstm_27_3/lstm_cell_138/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 19:25:25.251729: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_3/cond/Merge' id:63349 op device:{requested: '', assigned: ''} def:{{{node dropout_4_3/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_3/cond/Identity, dropout_4_3/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1066)\n",
      "(1514, 1066)\n",
      "(1644, 1066)\n",
      "(1764, 1066)\n",
      "(1836, 1066)\n",
      "(1699, 1066)\n",
      "(1369, 1066)\n",
      "(1766, 1066)\n",
      "(1631, 1066)\n",
      "(1692, 1066)\n",
      "(1550, 1066)\n",
      "(1920, 1066)\n",
      "(1739, 1066)\n",
      "(1788, 1066)\n",
      "(1812, 1066)\n",
      "(1788, 1066)\n",
      "(1788, 1066)\n",
      "(1668, 1066)\n",
      "(1872, 1066)\n",
      "{1: 6.02400462998145, 2: 6.09490022102002, 4: 10.0, 5: 5.601264245712188, 6: 5.745566200815446, 8: 8.743315804251296, 9: 8.830116557463207, 10: 7.374408153502923, 11: 7.876022807538199, 12: 8.56704333928213, 13: 6.38670064226194, 17: 9.204975870181336, 19: 8.454427519752215, 21: 8.127062313566853, 22: 1.0, 25: 7.536539296847565, 26: 6.076707465645105, 28: 5.294971534573275, 29: 2.4936309757597916}\n",
      "Train on 32325 samples, validate on 3610 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:29:05.209630: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_7/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32325/32325 [==============================] - ETA: 0s - loss: 9.8012\n",
      "Epoch 1: val_loss improved from inf to 1.37153, saving model to ./checkpoints/unknown_person_p27_21.h5\n",
      "32325/32325 [==============================] - 30s 916us/sample - loss: 9.8012 - val_loss: 1.3715\n",
      "Epoch 2/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.7564\n",
      "Epoch 2: val_loss improved from 1.37153 to 1.36869, saving model to ./checkpoints/unknown_person_p27_21.h5\n",
      "32325/32325 [==============================] - 21s 637us/sample - loss: 9.7564 - val_loss: 1.3687\n",
      "Epoch 3/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.6700\n",
      "Epoch 3: val_loss improved from 1.36869 to 1.36321, saving model to ./checkpoints/unknown_person_p27_21.h5\n",
      "32325/32325 [==============================] - 23s 723us/sample - loss: 9.6700 - val_loss: 1.3632\n",
      "Epoch 4/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.6526\n",
      "Epoch 4: val_loss improved from 1.36321 to 1.35300, saving model to ./checkpoints/unknown_person_p27_21.h5\n",
      "32325/32325 [==============================] - 23s 720us/sample - loss: 9.6526 - val_loss: 1.3530\n",
      "Epoch 5/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.6504\n",
      "Epoch 5: val_loss improved from 1.35300 to 1.34919, saving model to ./checkpoints/unknown_person_p27_21.h5\n",
      "32325/32325 [==============================] - 24s 728us/sample - loss: 9.6504 - val_loss: 1.3492\n",
      "Epoch 6/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.6229\n",
      "Epoch 6: val_loss did not improve from 1.34919\n",
      "32325/32325 [==============================] - 23s 716us/sample - loss: 9.6229 - val_loss: 1.3551\n",
      "Epoch 7/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5938\n",
      "Epoch 7: val_loss did not improve from 1.34919\n",
      "32325/32325 [==============================] - 23s 712us/sample - loss: 9.5938 - val_loss: 1.3563\n",
      "Epoch 8/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5710\n",
      "Epoch 8: val_loss improved from 1.34919 to 1.34724, saving model to ./checkpoints/unknown_person_p27_21.h5\n",
      "32325/32325 [==============================] - 23s 722us/sample - loss: 9.5710 - val_loss: 1.3472\n",
      "Epoch 9/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5624\n",
      "Epoch 9: val_loss improved from 1.34724 to 1.34554, saving model to ./checkpoints/unknown_person_p27_21.h5\n",
      "32325/32325 [==============================] - 24s 728us/sample - loss: 9.5624 - val_loss: 1.3455\n",
      "Epoch 10/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5651\n",
      "Epoch 10: val_loss did not improve from 1.34554\n",
      "32325/32325 [==============================] - 23s 719us/sample - loss: 9.5651 - val_loss: 1.3601\n",
      "Epoch 11/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5434\n",
      "Epoch 11: val_loss improved from 1.34554 to 1.34135, saving model to ./checkpoints/unknown_person_p27_21.h5\n",
      "32325/32325 [==============================] - 21s 635us/sample - loss: 9.5434 - val_loss: 1.3413\n",
      "Epoch 12/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5234\n",
      "Epoch 12: val_loss did not improve from 1.34135\n",
      "32325/32325 [==============================] - 20s 633us/sample - loss: 9.5234 - val_loss: 1.3510\n",
      "Epoch 13/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5017\n",
      "Epoch 13: val_loss did not improve from 1.34135\n",
      "32325/32325 [==============================] - 22s 694us/sample - loss: 9.5017 - val_loss: 1.3448\n",
      "Epoch 14/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5102\n",
      "Epoch 14: val_loss improved from 1.34135 to 1.33932, saving model to ./checkpoints/unknown_person_p27_21.h5\n",
      "32325/32325 [==============================] - 23s 726us/sample - loss: 9.5102 - val_loss: 1.3393\n",
      "Epoch 15/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.4933\n",
      "Epoch 15: val_loss did not improve from 1.33932\n",
      "32325/32325 [==============================] - 23s 723us/sample - loss: 9.4933 - val_loss: 1.3518\n",
      "Epoch 16/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.4873\n",
      "Epoch 16: val_loss did not improve from 1.33932\n",
      "32325/32325 [==============================] - 22s 673us/sample - loss: 9.4873 - val_loss: 1.3432\n",
      "Epoch 17/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.4913\n",
      "Epoch 17: val_loss improved from 1.33932 to 1.33590, saving model to ./checkpoints/unknown_person_p27_21.h5\n",
      "32325/32325 [==============================] - 20s 632us/sample - loss: 9.4913 - val_loss: 1.3359\n",
      "Epoch 18/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.4816\n",
      "Epoch 18: val_loss did not improve from 1.33590\n",
      "32325/32325 [==============================] - 21s 642us/sample - loss: 9.4816 - val_loss: 1.3421\n",
      "Epoch 19/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.4688\n",
      "Epoch 19: val_loss did not improve from 1.33590\n",
      "32325/32325 [==============================] - 23s 703us/sample - loss: 9.4688 - val_loss: 1.3452\n",
      "Epoch 20/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.4812\n",
      "Epoch 20: val_loss improved from 1.33590 to 1.33523, saving model to ./checkpoints/unknown_person_p27_21.h5\n",
      "32325/32325 [==============================] - 21s 649us/sample - loss: 9.4812 - val_loss: 1.3352\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:36:41.038350: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_9_4/lstm_cell_157/kernel/Assign' id:78236 op device:{requested: '', assigned: ''} def:{{{node lstm_9_4/lstm_cell_157/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_9_4/lstm_cell_157/kernel, lstm_9_4/lstm_cell_157/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 19:36:45.239630: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_3_4/lstm_cell_151/bias/v/Assign' id:84106 op device:{requested: '', assigned: ''} def:{{{node lstm_3_4/lstm_cell_151/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_3_4/lstm_cell_151/bias/v, lstm_3_4/lstm_cell_151/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 32325 samples, validate on 3610 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:36:51.959016: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:37:13.823872: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_8/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3774"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:37:36.309602: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.34628, saving model to ./checkpoints/unknown_person_baseline_2_p27_21.h5\n",
      "32325/32325 [==============================] - 32s 985us/sample - loss: 1.3774 - val_loss: 1.3463\n",
      "Epoch 2/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3734\n",
      "Epoch 2: val_loss did not improve from 1.34628\n",
      "32325/32325 [==============================] - 23s 715us/sample - loss: 1.3734 - val_loss: 1.3622\n",
      "Epoch 3/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3792\n",
      "Epoch 3: val_loss improved from 1.34628 to 1.34454, saving model to ./checkpoints/unknown_person_baseline_2_p27_21.h5\n",
      "32325/32325 [==============================] - 22s 672us/sample - loss: 1.3792 - val_loss: 1.3445\n",
      "Epoch 4/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3733\n",
      "Epoch 4: val_loss did not improve from 1.34454\n",
      "32325/32325 [==============================] - 22s 688us/sample - loss: 1.3733 - val_loss: 1.3456\n",
      "Epoch 5/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3738\n",
      "Epoch 5: val_loss did not improve from 1.34454\n",
      "32325/32325 [==============================] - 23s 715us/sample - loss: 1.3738 - val_loss: 1.3451\n",
      "Epoch 6/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3699\n",
      "Epoch 6: val_loss did not improve from 1.34454\n",
      "32325/32325 [==============================] - 22s 672us/sample - loss: 1.3699 - val_loss: 1.3479\n",
      "Epoch 7/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3675\n",
      "Epoch 7: val_loss improved from 1.34454 to 1.33809, saving model to ./checkpoints/unknown_person_baseline_2_p27_21.h5\n",
      "32325/32325 [==============================] - 21s 638us/sample - loss: 1.3675 - val_loss: 1.3381\n",
      "Epoch 8/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3698\n",
      "Epoch 8: val_loss did not improve from 1.33809\n",
      "32325/32325 [==============================] - 20s 627us/sample - loss: 1.3698 - val_loss: 1.3423\n",
      "Epoch 9/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3667\n",
      "Epoch 9: val_loss did not improve from 1.33809\n",
      "32325/32325 [==============================] - 20s 629us/sample - loss: 1.3667 - val_loss: 1.3405\n",
      "Epoch 10/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3679\n",
      "Epoch 10: val_loss improved from 1.33809 to 1.33324, saving model to ./checkpoints/unknown_person_baseline_2_p27_21.h5\n",
      "32325/32325 [==============================] - 21s 640us/sample - loss: 1.3679 - val_loss: 1.3332\n",
      "Epoch 11/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3670\n",
      "Epoch 11: val_loss did not improve from 1.33324\n",
      "32325/32325 [==============================] - 21s 655us/sample - loss: 1.3670 - val_loss: 1.3355\n",
      "Epoch 12/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3666\n",
      "Epoch 12: val_loss did not improve from 1.33324\n",
      "32325/32325 [==============================] - 22s 677us/sample - loss: 1.3666 - val_loss: 1.3344\n",
      "Epoch 13/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3662\n",
      "Epoch 13: val_loss did not improve from 1.33324\n",
      "32325/32325 [==============================] - 23s 709us/sample - loss: 1.3662 - val_loss: 1.3395\n",
      "Epoch 14/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3662\n",
      "Epoch 14: val_loss did not improve from 1.33324\n",
      "32325/32325 [==============================] - 22s 672us/sample - loss: 1.3662 - val_loss: 1.3448\n",
      "Epoch 15/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3630\n",
      "Epoch 15: val_loss improved from 1.33324 to 1.33231, saving model to ./checkpoints/unknown_person_baseline_2_p27_21.h5\n",
      "32325/32325 [==============================] - 22s 669us/sample - loss: 1.3630 - val_loss: 1.3323\n",
      "Epoch 16/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3615\n",
      "Epoch 16: val_loss did not improve from 1.33231\n",
      "32325/32325 [==============================] - 22s 688us/sample - loss: 1.3615 - val_loss: 1.3387\n",
      "Epoch 17/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3625\n",
      "Epoch 17: val_loss did not improve from 1.33231\n",
      "32325/32325 [==============================] - 22s 672us/sample - loss: 1.3625 - val_loss: 1.3441\n",
      "Epoch 18/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3622\n",
      "Epoch 18: val_loss improved from 1.33231 to 1.33161, saving model to ./checkpoints/unknown_person_baseline_2_p27_21.h5\n",
      "32325/32325 [==============================] - 20s 627us/sample - loss: 1.3622 - val_loss: 1.3316\n",
      "Epoch 19/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3607\n",
      "Epoch 19: val_loss improved from 1.33161 to 1.32761, saving model to ./checkpoints/unknown_person_baseline_2_p27_21.h5\n",
      "32325/32325 [==============================] - 20s 618us/sample - loss: 1.3607 - val_loss: 1.3276\n",
      "Epoch 20/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3597\n",
      "Epoch 20: val_loss did not improve from 1.32761\n",
      "32325/32325 [==============================] - 21s 653us/sample - loss: 1.3597 - val_loss: 1.3444\n",
      "Train on 32325 samples, validate on 3610 samples\n",
      "Epoch 1/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3599\n",
      "Epoch 1: val_loss improved from inf to 1.33318, saving model to ./checkpoints/unknown_person_baseline_p27_22.h5\n",
      "32325/32325 [==============================] - 23s 722us/sample - loss: 1.3599 - val_loss: 1.3332\n",
      "Epoch 2/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3572\n",
      "Epoch 2: val_loss did not improve from 1.33318\n",
      "32325/32325 [==============================] - 20s 633us/sample - loss: 1.3572 - val_loss: 1.3395\n",
      "Epoch 3/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3570\n",
      "Epoch 3: val_loss improved from 1.33318 to 1.33047, saving model to ./checkpoints/unknown_person_baseline_p27_22.h5\n",
      "32325/32325 [==============================] - 21s 636us/sample - loss: 1.3570 - val_loss: 1.3305\n",
      "Epoch 4/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3547\n",
      "Epoch 4: val_loss improved from 1.33047 to 1.32738, saving model to ./checkpoints/unknown_person_baseline_p27_22.h5\n",
      "32325/32325 [==============================] - 21s 643us/sample - loss: 1.3547 - val_loss: 1.3274\n",
      "Epoch 5/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3539\n",
      "Epoch 5: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 699us/sample - loss: 1.3539 - val_loss: 1.3351\n",
      "Epoch 6/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3519\n",
      "Epoch 6: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 22s 694us/sample - loss: 1.3519 - val_loss: 1.3291\n",
      "Epoch 7/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3541\n",
      "Epoch 7: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 633us/sample - loss: 1.3541 - val_loss: 1.3334\n",
      "Epoch 8/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3564\n",
      "Epoch 8: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 625us/sample - loss: 1.3564 - val_loss: 1.3327\n",
      "Epoch 9/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3837\n",
      "Epoch 9: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 622us/sample - loss: 1.3837 - val_loss: 1.3274\n",
      "Epoch 10/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3668\n",
      "Epoch 10: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 705us/sample - loss: 1.3668 - val_loss: 1.3420\n",
      "Epoch 11/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4876\n",
      "Epoch 11: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 698us/sample - loss: 1.4876 - val_loss: 1.3558\n",
      "Epoch 12/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3977\n",
      "Epoch 12: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 21s 636us/sample - loss: 1.3977 - val_loss: 1.3476\n",
      "Epoch 13/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4113\n",
      "Epoch 13: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 632us/sample - loss: 1.4113 - val_loss: 1.3504\n",
      "Epoch 14/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4036\n",
      "Epoch 14: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 22s 684us/sample - loss: 1.4036 - val_loss: 1.3491\n",
      "Epoch 15/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3964\n",
      "Epoch 15: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 21s 661us/sample - loss: 1.3964 - val_loss: 1.3522\n",
      "Epoch 16/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4019\n",
      "Epoch 16: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 626us/sample - loss: 1.4019 - val_loss: 1.3510\n",
      "Epoch 17/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4062\n",
      "Epoch 17: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 22s 678us/sample - loss: 1.4062 - val_loss: 1.3504\n",
      "Epoch 18/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4077\n",
      "Epoch 18: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 710us/sample - loss: 1.4077 - val_loss: 1.3573\n",
      "Epoch 19/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4365\n",
      "Epoch 19: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 699us/sample - loss: 1.4365 - val_loss: 1.3771\n",
      "Epoch 20/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4226\n",
      "Epoch 20: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 720us/sample - loss: 1.4226 - val_loss: 1.3698\n",
      "Epoch 21/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4328\n",
      "Epoch 21: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 717us/sample - loss: 1.4328 - val_loss: 1.3715\n",
      "Epoch 22/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4503\n",
      "Epoch 22: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 22s 682us/sample - loss: 1.4503 - val_loss: 1.3789\n",
      "Epoch 23/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4356\n",
      "Epoch 23: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 21s 636us/sample - loss: 1.4356 - val_loss: 1.3740\n",
      "Epoch 24/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4280\n",
      "Epoch 24: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 633us/sample - loss: 1.4280 - val_loss: 1.3828\n",
      "Epoch 25/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4301\n",
      "Epoch 25: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 706us/sample - loss: 1.4301 - val_loss: 1.3735\n",
      "Epoch 26/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4250\n",
      "Epoch 26: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 714us/sample - loss: 1.4250 - val_loss: 1.3586\n",
      "Epoch 27/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4222\n",
      "Epoch 27: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 714us/sample - loss: 1.4222 - val_loss: 1.3690\n",
      "Epoch 28/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4173\n",
      "Epoch 28: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 716us/sample - loss: 1.4173 - val_loss: 1.3607\n",
      "Epoch 29/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4182\n",
      "Epoch 29: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 716us/sample - loss: 1.4182 - val_loss: 1.3649\n",
      "Epoch 30/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4129\n",
      "Epoch 30: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 21s 661us/sample - loss: 1.4129 - val_loss: 1.3597\n",
      "Epoch 31/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4112\n",
      "Epoch 31: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 22s 692us/sample - loss: 1.4112 - val_loss: 1.3585\n",
      "Epoch 32/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4105\n",
      "Epoch 32: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 628us/sample - loss: 1.4105 - val_loss: 1.3576\n",
      "Epoch 33/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4088\n",
      "Epoch 33: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 21s 649us/sample - loss: 1.4088 - val_loss: 1.3693\n",
      "Epoch 34/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4057\n",
      "Epoch 34: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 629us/sample - loss: 1.4057 - val_loss: 1.3583\n",
      "Epoch 35/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4059\n",
      "Epoch 35: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 21s 652us/sample - loss: 1.4059 - val_loss: 1.3601\n",
      "Epoch 36/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4049\n",
      "Epoch 36: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 633us/sample - loss: 1.4049 - val_loss: 1.3561\n",
      "Epoch 37/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4031\n",
      "Epoch 37: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 632us/sample - loss: 1.4031 - val_loss: 1.3512\n",
      "Epoch 38/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4058\n",
      "Epoch 38: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 22s 686us/sample - loss: 1.4058 - val_loss: 1.3580\n",
      "Epoch 39/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4001\n",
      "Epoch 39: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 707us/sample - loss: 1.4001 - val_loss: 1.3556\n",
      "Epoch 40/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4005\n",
      "Epoch 40: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 709us/sample - loss: 1.4005 - val_loss: 1.3543\n",
      "Epoch 41/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4011\n",
      "Epoch 41: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 21s 640us/sample - loss: 1.4011 - val_loss: 1.3528\n",
      "Epoch 42/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3967\n",
      "Epoch 42: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 616us/sample - loss: 1.3967 - val_loss: 1.3544\n",
      "Epoch 43/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.4005\n",
      "Epoch 43: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 621us/sample - loss: 1.4005 - val_loss: 1.3554\n",
      "Epoch 44/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3951\n",
      "Epoch 44: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 21s 645us/sample - loss: 1.3951 - val_loss: 1.3561\n",
      "Epoch 45/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3953\n",
      "Epoch 45: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 21s 646us/sample - loss: 1.3953 - val_loss: 1.3521\n",
      "Epoch 46/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3959\n",
      "Epoch 46: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 21s 662us/sample - loss: 1.3959 - val_loss: 1.3550\n",
      "Epoch 47/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3924\n",
      "Epoch 47: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 22s 691us/sample - loss: 1.3924 - val_loss: 1.3538\n",
      "Epoch 48/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3921\n",
      "Epoch 48: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 20s 633us/sample - loss: 1.3921 - val_loss: 1.3538\n",
      "Epoch 49/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3890\n",
      "Epoch 49: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 21s 660us/sample - loss: 1.3890 - val_loss: 1.3508\n",
      "Epoch 50/50\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3901\n",
      "Epoch 50: val_loss did not improve from 1.32738\n",
      "32325/32325 [==============================] - 23s 714us/sample - loss: 1.3901 - val_loss: 1.3517\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:02:36.540793: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_6_5/lstm_cell_191/kernel/Assign' id:97116 op device:{requested: '', assigned: ''} def:{{{node lstm_6_5/lstm_cell_191/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_6_5/lstm_cell_191/kernel, lstm_6_5/lstm_cell_191/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:02:41.574626: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_1_5/lstm_cell_186/bias/v/Assign' id:103436 op device:{requested: '', assigned: ''} def:{{{node lstm_1_5/lstm_cell_186/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_1_5/lstm_cell_186/bias/v, lstm_1_5/lstm_cell_186/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:02:46.478225: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_5/cond/Merge' id:102107 op device:{requested: '', assigned: ''} def:{{{node dropout_4_5/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_5/cond/Identity, dropout_4_5/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1066)\n",
      "(1514, 1066)\n",
      "(1644, 1066)\n",
      "(1764, 1066)\n",
      "(1836, 1066)\n",
      "(1699, 1066)\n",
      "(1369, 1066)\n",
      "(1766, 1066)\n",
      "(1631, 1066)\n",
      "(1692, 1066)\n",
      "(1550, 1066)\n",
      "(1920, 1066)\n",
      "(1739, 1066)\n",
      "(1788, 1066)\n",
      "(1812, 1066)\n",
      "(1788, 1066)\n",
      "(1788, 1066)\n",
      "(1668, 1066)\n",
      "(1872, 1066)\n",
      "{1: 5.982389503396442, 2: 6.011822923255076, 4: 10.0, 5: 5.510068435228222, 6: 5.543582630631348, 8: 8.615531666742848, 9: 8.711675156041732, 10: 7.311511666033535, 11: 7.782947348544405, 12: 8.55270407429122, 13: 6.3911158768047995, 17: 9.16349145774854, 19: 8.439539958946789, 21: 8.134993384733118, 22: 1.0, 25: 7.4643824579995055, 26: 6.104929313619313, 28: 5.151121957187419, 29: 1.948563751197933}\n",
      "Train on 32325 samples, validate on 3610 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:07:34.373166: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_9/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5423\n",
      "Epoch 1: val_loss improved from inf to 1.34586, saving model to ./checkpoints/unknown_person_p27_22.h5\n",
      "32325/32325 [==============================] - 37s 1ms/sample - loss: 9.5423 - val_loss: 1.3459\n",
      "Epoch 2/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5462\n",
      "Epoch 2: val_loss improved from 1.34586 to 1.34094, saving model to ./checkpoints/unknown_person_p27_22.h5\n",
      "32325/32325 [==============================] - 23s 727us/sample - loss: 9.5462 - val_loss: 1.3409\n",
      "Epoch 3/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.7007\n",
      "Epoch 3: val_loss improved from 1.34094 to 1.34013, saving model to ./checkpoints/unknown_person_p27_22.h5\n",
      "32325/32325 [==============================] - 22s 669us/sample - loss: 9.7007 - val_loss: 1.3401\n",
      "Epoch 4/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.5290\n",
      "Epoch 4: val_loss did not improve from 1.34013\n",
      "32325/32325 [==============================] - 20s 631us/sample - loss: 9.5290 - val_loss: 1.3424\n",
      "Epoch 5/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.4423\n",
      "Epoch 5: val_loss did not improve from 1.34013\n",
      "32325/32325 [==============================] - 23s 698us/sample - loss: 9.4423 - val_loss: 1.3539\n",
      "Epoch 6/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.4210\n",
      "Epoch 6: val_loss did not improve from 1.34013\n",
      "32325/32325 [==============================] - 23s 721us/sample - loss: 9.4210 - val_loss: 1.3540\n",
      "Epoch 7/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3977\n",
      "Epoch 7: val_loss did not improve from 1.34013\n",
      "32325/32325 [==============================] - 23s 719us/sample - loss: 9.3977 - val_loss: 1.3431\n",
      "Epoch 8/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3950\n",
      "Epoch 8: val_loss improved from 1.34013 to 1.33968, saving model to ./checkpoints/unknown_person_p27_22.h5\n",
      "32325/32325 [==============================] - 23s 726us/sample - loss: 9.3950 - val_loss: 1.3397\n",
      "Epoch 9/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3836\n",
      "Epoch 9: val_loss did not improve from 1.33968\n",
      "32325/32325 [==============================] - 23s 714us/sample - loss: 9.3836 - val_loss: 1.3476\n",
      "Epoch 10/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3712\n",
      "Epoch 10: val_loss did not improve from 1.33968\n",
      "32325/32325 [==============================] - 21s 637us/sample - loss: 9.3712 - val_loss: 1.3412\n",
      "Epoch 11/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3584\n",
      "Epoch 11: val_loss did not improve from 1.33968\n",
      "32325/32325 [==============================] - 21s 639us/sample - loss: 9.3584 - val_loss: 1.3425\n",
      "Epoch 12/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3433\n",
      "Epoch 12: val_loss did not improve from 1.33968\n",
      "32325/32325 [==============================] - 20s 627us/sample - loss: 9.3433 - val_loss: 1.3438\n",
      "Epoch 13/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3297\n",
      "Epoch 13: val_loss improved from 1.33968 to 1.33875, saving model to ./checkpoints/unknown_person_p27_22.h5\n",
      "32325/32325 [==============================] - 22s 682us/sample - loss: 9.3297 - val_loss: 1.3388\n",
      "Epoch 14/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3412\n",
      "Epoch 14: val_loss did not improve from 1.33875\n",
      "32325/32325 [==============================] - 21s 654us/sample - loss: 9.3412 - val_loss: 1.3412\n",
      "Epoch 15/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3386\n",
      "Epoch 15: val_loss did not improve from 1.33875\n",
      "32325/32325 [==============================] - 23s 706us/sample - loss: 9.3386 - val_loss: 1.3496\n",
      "Epoch 16/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3343\n",
      "Epoch 16: val_loss did not improve from 1.33875\n",
      "32325/32325 [==============================] - 23s 715us/sample - loss: 9.3343 - val_loss: 1.3479\n",
      "Epoch 17/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3228\n",
      "Epoch 17: val_loss did not improve from 1.33875\n",
      "32325/32325 [==============================] - 23s 699us/sample - loss: 9.3228 - val_loss: 1.3417\n",
      "Epoch 18/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3123\n",
      "Epoch 18: val_loss did not improve from 1.33875\n",
      "32325/32325 [==============================] - 20s 629us/sample - loss: 9.3123 - val_loss: 1.3445\n",
      "Epoch 19/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.3066\n",
      "Epoch 19: val_loss did not improve from 1.33875\n",
      "32325/32325 [==============================] - 20s 633us/sample - loss: 9.3066 - val_loss: 1.3588\n",
      "Epoch 20/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 9.2804\n",
      "Epoch 20: val_loss did not improve from 1.33875\n",
      "32325/32325 [==============================] - 20s 630us/sample - loss: 9.2804 - val_loss: 1.3487\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:15:09.513839: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_35_6/lstm_cell_257/recurrent_kernel/Assign' id:121176 op device:{requested: '', assigned: ''} def:{{{node lstm_35_6/lstm_cell_257/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_35_6/lstm_cell_257/recurrent_kernel, lstm_35_6/lstm_cell_257/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:15:15.530210: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_3_6/kernel/v/Assign' id:123400 op device:{requested: '', assigned: ''} def:{{{node dense_3_6/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_3_6/kernel/v, dense_3_6/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 32325 samples, validate on 3610 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:15:24.434214: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:15:55.798515: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_10/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3541"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:16:20.736729: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.33626, saving model to ./checkpoints/unknown_person_baseline_2_p27_22.h5\n",
      "32325/32325 [==============================] - 46s 1ms/sample - loss: 1.3541 - val_loss: 1.3363\n",
      "Epoch 2/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3536\n",
      "Epoch 2: val_loss improved from 1.33626 to 1.33574, saving model to ./checkpoints/unknown_person_baseline_2_p27_22.h5\n",
      "32325/32325 [==============================] - 21s 641us/sample - loss: 1.3536 - val_loss: 1.3357\n",
      "Epoch 3/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3863\n",
      "Epoch 3: val_loss did not improve from 1.33574\n",
      "32325/32325 [==============================] - 21s 637us/sample - loss: 1.3863 - val_loss: 1.3378\n",
      "Epoch 4/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3672\n",
      "Epoch 4: val_loss improved from 1.33574 to 1.32918, saving model to ./checkpoints/unknown_person_baseline_2_p27_22.h5\n",
      "32325/32325 [==============================] - 21s 636us/sample - loss: 1.3672 - val_loss: 1.3292\n",
      "Epoch 5/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3590\n",
      "Epoch 5: val_loss did not improve from 1.32918\n",
      "32325/32325 [==============================] - 20s 632us/sample - loss: 1.3590 - val_loss: 1.3308\n",
      "Epoch 6/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3572\n",
      "Epoch 6: val_loss did not improve from 1.32918\n",
      "32325/32325 [==============================] - 23s 706us/sample - loss: 1.3572 - val_loss: 1.3372\n",
      "Epoch 7/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3566\n",
      "Epoch 7: val_loss did not improve from 1.32918\n",
      "32325/32325 [==============================] - 23s 718us/sample - loss: 1.3566 - val_loss: 1.3307\n",
      "Epoch 8/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3553\n",
      "Epoch 8: val_loss did not improve from 1.32918\n",
      "32325/32325 [==============================] - 21s 652us/sample - loss: 1.3553 - val_loss: 1.3356\n",
      "Epoch 9/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3508\n",
      "Epoch 9: val_loss did not improve from 1.32918\n",
      "32325/32325 [==============================] - 23s 705us/sample - loss: 1.3508 - val_loss: 1.3316\n",
      "Epoch 10/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3499\n",
      "Epoch 10: val_loss improved from 1.32918 to 1.32412, saving model to ./checkpoints/unknown_person_baseline_2_p27_22.h5\n",
      "32325/32325 [==============================] - 21s 654us/sample - loss: 1.3499 - val_loss: 1.3241\n",
      "Epoch 11/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3539\n",
      "Epoch 11: val_loss did not improve from 1.32412\n",
      "32325/32325 [==============================] - 22s 674us/sample - loss: 1.3539 - val_loss: 1.3281\n",
      "Epoch 12/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3508\n",
      "Epoch 12: val_loss did not improve from 1.32412\n",
      "32325/32325 [==============================] - 23s 718us/sample - loss: 1.3508 - val_loss: 1.3288\n",
      "Epoch 13/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3493\n",
      "Epoch 13: val_loss did not improve from 1.32412\n",
      "32325/32325 [==============================] - 23s 723us/sample - loss: 1.3493 - val_loss: 1.3371\n",
      "Epoch 14/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3510\n",
      "Epoch 14: val_loss did not improve from 1.32412\n",
      "32325/32325 [==============================] - 22s 679us/sample - loss: 1.3510 - val_loss: 1.3321\n",
      "Epoch 15/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3502\n",
      "Epoch 15: val_loss did not improve from 1.32412\n",
      "32325/32325 [==============================] - 21s 636us/sample - loss: 1.3502 - val_loss: 1.3286\n",
      "Epoch 16/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3498\n",
      "Epoch 16: val_loss did not improve from 1.32412\n",
      "32325/32325 [==============================] - 22s 692us/sample - loss: 1.3498 - val_loss: 1.3466\n",
      "Epoch 17/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3482\n",
      "Epoch 17: val_loss did not improve from 1.32412\n",
      "32325/32325 [==============================] - 20s 629us/sample - loss: 1.3482 - val_loss: 1.3389\n",
      "Epoch 18/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3498\n",
      "Epoch 18: val_loss did not improve from 1.32412\n",
      "32325/32325 [==============================] - 20s 629us/sample - loss: 1.3498 - val_loss: 1.3375\n",
      "Epoch 19/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3494\n",
      "Epoch 19: val_loss did not improve from 1.32412\n",
      "32325/32325 [==============================] - 20s 629us/sample - loss: 1.3494 - val_loss: 1.3340\n",
      "Epoch 20/20\n",
      "32325/32325 [==============================] - ETA: 0s - loss: 1.3510\n",
      "Epoch 20: val_loss did not improve from 1.32412\n",
      "32325/32325 [==============================] - 20s 628us/sample - loss: 1.3510 - val_loss: 1.3297\n"
     ]
    }
   ],
   "source": [
    "person_nums = [1,2,4,5,6,8,9,10,11,12,13,17,19,21,22,25,26,27,28,29]\n",
    "## Build the baseline model for emotion recognition with dropout layers\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, SimpleRNN, LSTM, Conv2D, Flatten, MaxPooling2D, GRU, AveragePooling2D, Dropout, Lambda\n",
    "from tensorflow.keras.callbacks import ModelCheckpoint\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "def individual_model(features_list):\n",
    "    input_layers = []\n",
    "    hidden_layers = []\n",
    "    combined_feature = np.empty((len(features_list[0]),0))\n",
    "    for i, feature in enumerate(features_list):\n",
    "        \n",
    "        if len(feature.shape) == 3:\n",
    "            input_i = Input(shape=(feature.shape[1], feature.shape[2]))\n",
    "            input_layers.append(input_i)\n",
    "\n",
    "            hidden_i = input_i[:,:,:,None]\n",
    "            hidden_i = Conv2D(32, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(16, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(8, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(4, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Flatten()(hidden_i)\n",
    "\n",
    "            hidden_layers.append(hidden_i)\n",
    "        elif feature.shape[1] <20:\n",
    "            combined_feature = np.hstack((combined_feature, feature))\n",
    "            \n",
    "        else:  # For series features\n",
    "            input_i = Input(shape=(feature.shape[1],))\n",
    "            input_layers.append(input_i)\n",
    "            hidden_i = Lambda(lambda x: x[:, :, None])(input_i)  # Add a new dimension\n",
    "            hidden_i = LSTM(4)(hidden_i)\n",
    "            hidden_layers.append(hidden_i)\n",
    "    input_i = Input(shape=(combined_feature.shape[1],))\n",
    "    input_layers.append(input_i)\n",
    "    dense_num = np.max((1, int(combined_feature.shape[1]/2)))\n",
    "    hidden_i = Dense(dense_num, activation='relu')(input_i)\n",
    "    hidden_layers.append(hidden_i)\n",
    "    print(combined_feature.shape)\n",
    "    concat_layer = concatenate(hidden_layers)\n",
    "    h = Dropout(0.2)(concat_layer)\n",
    "    h = Dense(64, activation='relu')(h)\n",
    "    h = Dense(32, activation='relu')(h)\n",
    "    output_layer = Dense(2)(h)\n",
    "    model = Model(inputs=input_layers, outputs=output_layer)\n",
    "    model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "    return model\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.callbacks import Callback\n",
    "import numpy as np\n",
    "\n",
    "class PruningCallback(tf.keras.callbacks.Callback):\n",
    "    def __init__(self, initial_pruning_factor=0.1, final_pruning_factor=0.5, start_epoch=0, end_epoch=None, frequency=1):\n",
    "        super(PruningCallback, self).__init__()\n",
    "        self.initial_pruning_factor = initial_pruning_factor\n",
    "        self.final_pruning_factor = final_pruning_factor\n",
    "        self.start_epoch = start_epoch\n",
    "        self.end_epoch = end_epoch if end_epoch is not None else np.inf\n",
    "        self.frequency = frequency\n",
    "        self.pruned_weights = {}\n",
    "        self.layer_importance = {}\n",
    "\n",
    "    def get_pruning_factor(self, epoch):\n",
    "        if epoch < self.start_epoch:\n",
    "            return 0\n",
    "        if epoch > self.end_epoch:\n",
    "            return self.final_pruning_factor\n",
    "        return self.initial_pruning_factor + (self.final_pruning_factor - self.initial_pruning_factor) * (epoch - self.start_epoch) / (self.end_epoch - self.start_epoch)\n",
    "\n",
    "    def on_train_begin(self, logs=None):\n",
    "        total_weight_magnitude = 0\n",
    "        for layer in self.model.layers:\n",
    "            if hasattr(layer, 'get_weights'):\n",
    "                weights = layer.get_weights()\n",
    "                layer_norm = sum(np.linalg.norm(w) for w in weights)\n",
    "                total_weight_magnitude += layer_norm\n",
    "                self.layer_importance[layer.name] = layer_norm\n",
    "    \n",
    "        # Normalize the layer importance values so they sum up to 1\n",
    "        for layer_name in self.layer_importance:\n",
    "            self.layer_importance[layer_name] /= total_weight_magnitude\n",
    "    # def on_train_begin(self, logs=None):\n",
    "    #     total_weight_magnitude = sum([np.linalg.norm(layer.get_weights()) for layer in self.model.layers if hasattr(layer, 'get_weights')])\n",
    "    #     for layer in self.model.layers:\n",
    "    #         if hasattr(layer, 'get_weights'):\n",
    "    #             self.layer_importance[layer.name] = np.linalg.norm(layer.get_weights()) / total_weight_magnitude\n",
    "\n",
    "    def get_layer_pruning_factor(self, layer_name, global_pruning_factor):\n",
    "        if layer_name in self.layer_importance:\n",
    "            importance = self.layer_importance[layer_name]\n",
    "            adjusted_pruning_factor = global_pruning_factor * (1 - importance)\n",
    "            return min(adjusted_pruning_factor, 1)  # Ensure the pruning factor is not greater than 1\n",
    "        return global_pruning_factor\n",
    "    def prune_weights(self, layer, global_pruning_factor):\n",
    "        \n",
    "        weights = layer.get_weights()\n",
    "        layer_name = layer.name\n",
    "        pruning_factor = self.get_layer_pruning_factor(layer_name, global_pruning_factor)\n",
    "\n",
    "        if layer_name not in self.pruned_weights:\n",
    "            self.pruned_weights[layer_name] = [np.zeros_like(w, dtype=bool) for w in weights]\n",
    "        \n",
    "        for i in range(len(weights)):\n",
    "            weight = weights[i]\n",
    "            # print(weight.shape)\n",
    "            # print(weight.size)\n",
    "            if weight.ndim > 1:  # Only prune dense or convolutional layers\n",
    "                unpruned_weights = np.logical_not(self.pruned_weights[layer_name][i])\n",
    "                num_unpruned = np.sum(unpruned_weights)\n",
    "                num_pruning = min(num_unpruned, int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i]))\n",
    "                num_pruning = int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i])\n",
    "                if num_pruning > 0:\n",
    "                    unpruned_flat_indices = np.flatnonzero(unpruned_weights)\n",
    "                    abs_unpruned_weights = np.abs(weight[unpruned_weights])\n",
    "                    pruning_flat_indices = np.argpartition(abs_unpruned_weights, num_pruning)[:num_pruning]\n",
    "                    \n",
    "                    indices = np.unravel_index(pruning_flat_indices, weight.shape)\n",
    "                    self.pruned_weights[layer_name][i][indices] = True\n",
    "\n",
    "                weights[i] = weights[i]*(~self.pruned_weights[layer_name][i])\n",
    "                \n",
    "        layer.set_weights(weights)\n",
    "\n",
    "    def on_epoch_end(self, epoch, logs=None):\n",
    "        if (epoch - self.start_epoch) % self.frequency != 0:\n",
    "            return\n",
    "\n",
    "        pruning_factor = self.get_pruning_factor(epoch)\n",
    "        for layer in self.model.layers:\n",
    "            if isinstance(layer, (tf.keras.layers.Dense, tf.keras.layers.Conv2D)):\n",
    "                self.prune_weights(layer, pruning_factor)\n",
    "from feature_emotion import feature_extract, split_data, label_unique_tuples, split_data_unknown\n",
    "gts, sensor_nums, walk_nums, trace_nums, people_nums, spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features = feature_extract(person_nums)\n",
    "\n",
    "walk_nums_all = np.squeeze(walk_nums)\n",
    "trace_nums_all = np.squeeze(trace_nums)\n",
    "people_nums_all = np.squeeze(people_nums)\n",
    "\n",
    "## 0: train, 1: validation 2: test\n",
    "\n",
    "test_person_id = [27]\n",
    "flag_tr_val_te = split_data_unknown(test_person_id, walk_nums_all, trace_nums_all, people_nums_all, person_nums, rand_seed = 42)\n",
    "## Data Normalization before training ans testing\n",
    "import tensorflow as tf\n",
    "tf.compat.v1.disable_v2_behavior()\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, LSTM\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scalers = []\n",
    "X_train_normalized = []\n",
    "X_val_normalized = []\n",
    "X_test_normalized = []\n",
    "train_idx = np.where(flag_tr_val_te ==0)[0]\n",
    "np.random.shuffle(train_idx)\n",
    "val_idx = np.where(flag_tr_val_te ==1)[0]\n",
    "test_idx = np.where(flag_tr_val_te ==2)[0]\n",
    "\n",
    "for i, feature in enumerate([spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features]):\n",
    "    scaler = StandardScaler()\n",
    "    if len(feature.shape)==2:\n",
    "        X_train_i = feature[train_idx,:]\n",
    "        X_val_i = feature[val_idx,:]\n",
    "        X_test_i = feature[test_idx,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i)\n",
    "        scalers.append(scaler)\n",
    "    else:\n",
    "        X_train_i = feature[train_idx,:,:]\n",
    "        X_val_i = feature[val_idx,:,:]\n",
    "        X_test_i = feature[test_idx,:,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i.reshape(X_train_i.shape[0], -1)).reshape(X_train_i.shape)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i.reshape(X_val_i.shape[0], -1)).reshape(X_val_i.shape)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i.reshape(X_test_i.shape[0], -1)).reshape(X_test_i.shape)\n",
    "        scalers.append(scaler)\n",
    "    X_train_normalized.append(X_train_normalized_i)\n",
    "    X_val_normalized.append(X_val_normalized_i)\n",
    "    X_test_normalized.append(X_test_normalized_i)\n",
    "y_train = gts[train_idx,:]\n",
    "y_val = gts[val_idx,:]\n",
    "y_test = gts[test_idx,:]\n",
    "X_train_normalized_new = []\n",
    "combined_feature = np.empty((len(X_train_normalized[0]),0))\n",
    "for feature in X_train_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_train_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_train_normalized_new.append(feature)\n",
    "X_train_normalized_new.append(combined_feature)\n",
    "\n",
    "X_val_normalized_new = []\n",
    "combined_feature = np.empty((len(X_val_normalized[0]),0))\n",
    "for feature in X_val_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_val_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_val_normalized_new.append(feature)\n",
    "X_val_normalized_new.append(combined_feature)\n",
    "\n",
    "X_test_normalized_new = []\n",
    "combined_feature = np.empty((len(X_test_normalized[0]),0))\n",
    "for feature in X_test_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_test_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_test_normalized_new.append(feature)\n",
    "X_test_normalized_new.append(combined_feature)\n",
    "\n",
    "model = individual_model(X_train_normalized)\n",
    "model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "num_epochs = 50\n",
    "\n",
    "\n",
    "# Gradually prune weights in dense and convolutional layers from 10% to 50% over the course of training, starting from epoch 0.\n",
    "\n",
    "\n",
    "rates = [0.4, 0.5, 0.6]\n",
    "idx = 20\n",
    "for r in rates:\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=0.01, final_pruning_factor=r, start_epoch=5, end_epoch=20, frequency=1)\n",
    "    model.fit(X_train_normalized_new, y_train, epochs=num_epochs, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import tensorflow as tf\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    from tensorflow.keras.models import Model\n",
    "    layers = model.layers\n",
    "    second_last_layer_output = layers[-4].output\n",
    "    feature_extractor_model = Model(inputs=model.input, outputs=second_last_layer_output)\n",
    "    train_features = feature_extractor_model.predict(X_train_normalized_new)\n",
    "    test_features = feature_extractor_model.predict(X_test_normalized_new)\n",
    "    \n",
    "    p_train = people_nums[train_idx,:]\n",
    "    p_val = people_nums[val_idx,:]\n",
    "    p_test = people_nums[test_idx,:]\n",
    "    ## Calculate the distance between test person and training person\n",
    "    def euclidean_distance(a, b):\n",
    "        return np.sqrt(np.sum((a - b) ** 2, axis=1))\n",
    "    \n",
    "    distance_dict = {}\n",
    "    for ii in range(len(person_nums)):\n",
    "        if person_nums[ii] == test_person_id[0]:\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            ind = np.where(p_train ==person_nums[ii])[0]\n",
    "            tmp_train_features = train_features[ind, :]\n",
    "            distances = np.array([euclidean_distance(train_sample, test_features) for train_sample in tmp_train_features])\n",
    "            print(distances.shape)\n",
    "            average_distances = np.mean(distances, axis=1)\n",
    "    \n",
    "            # Step 4: Find the overall average distance\n",
    "            overall_average_distance = np.mean(average_distances)\n",
    "            distance_dict[person_nums[ii]] = overall_average_distance\n",
    "    \n",
    "    \n",
    "    def normalize_to_weights(distance_dict):\n",
    "        distances = np.array(list(distance_dict.values()))\n",
    "        # Handle the case where a distance is zero to avoid division by zero\n",
    "        distances = np.clip(distances, a_min=1e-10, a_max=None)\n",
    "        weights = 1 / distances\n",
    "        normalized_weights = weights\n",
    "        # normalized_weights = weights / sum(weights)\n",
    "        # print(sum(weights))\n",
    "        # print(sum(normalized_weights))\n",
    "        # Assign the normalized weights back to the dictionary\n",
    "        normalized_weight_dict = dict(zip(distance_dict.keys(), normalized_weights))\n",
    "        return normalized_weight_dict\n",
    "    def scale_dict_values(my_dict):\n",
    "        scaled_dict = my_dict.copy()\n",
    "        min_val = min(scaled_dict.values())\n",
    "        max_val = max(scaled_dict.values())\n",
    "        \n",
    "        for key in scaled_dict:\n",
    "            scaled_dict[key] = 1 + 9 * (scaled_dict[key] - min_val) / (max_val - min_val)\n",
    "        \n",
    "        return scaled_dict\n",
    "    weights_dict = normalize_to_weights(distance_dict)\n",
    "    weights_dict = scale_dict_values(weights_dict)\n",
    "    print(weights_dict)\n",
    "    \n",
    "    w_train = np.zeros_like(p_train)\n",
    "    for i in range(len(w_train)):\n",
    "        w_train[i] = weights_dict[int(p_train[i])]\n",
    "    \n",
    "    w_train = np.squeeze(w_train)\n",
    "    \n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,sample_weight= w_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    model_name = './checkpoints/unknown_person_baseline_2_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    idx += 1\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71b59e4-6581-4ea8-994d-a02a6654ab6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
